src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -265,6 +269,17 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     if (this.props.showTrafficAnimation) {
       this.trafficRenderer.start();
     }
+
+    // Asynchronously fetch health
+    cy.nodes().forEach(ele => {
```

Comment:
I'm concerned that multiple layout changes will keep updating this data, even if our refresh interval is every 15 seconds or so.
Can we validate that we are only updating these fields when `this.props.elements` change?

Reply:
Hmm... But this.props.elements may not change on graph refresh, no? The health should be IMO updated at every refresh interval.

Reply:
The contents might not change, but this will be true after a graph refresh
`this.props.elements !== prevprops.elements`